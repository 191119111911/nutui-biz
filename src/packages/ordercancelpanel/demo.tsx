import React, { useState, CSSProperties } from "react";
import { OrderCancelPanel, IreasonsObject } from "./ordercancelpanel";
import { Cell, Button, ButtonProps, TextAreaProps } from "@nutui/nutui-react";
import { useTranslate } from "../../sites/assets/locale";

interface T {
  basic: string;
  tipsText: string;
  otherText: string;
  cellTitle: string;
  submitText: string;
  checkboxText: string;
  cancelText: string;
  cancelReasonTitle: string;
  reasonTitle: string;
  tipsTitle: string;
  textareaPlaceholder: string;
  warmTips: Array<string>;
  cancelReason: Array<IreasonsObject>;
  otherReason: Array<IreasonsObject>;
}

const CouponDemo = () => {
  const [translated] = useTranslate<T>({
    "zh-CN": {
      basic: "基本用法",
      tipsText: "带有温馨提示的组件",
      otherText: "带有其它原因选项的组件",
      cancelText: "可取消已选中的原因",
      checkboxText: "checkbox框选择在前面",
      cellTitle: "显示弹窗",
      submitText: "确认",
      cancelReasonTitle: "退款原因",
      reasonTitle: "请选择取消订单原因",
      tipsTitle: "温馨提示",
      textareaPlaceholder: "请输入内容",
      warmTips: [
        "1. 限时特价、预约资格等购买优惠可能一并取消",
        "2. 如遇订单拆分，京券将换成同价值京豆返还",
        "3. 支付券不予返还；支付优惠一并取消",
        "4. 订单一旦取消，无法恢复",
      ],
      cancelReason: [
        {
          key: "reasons1",
          value: "商品无货",
        },
        {
          key: "reasons2",
          value: "发货时间问题",
        },
        {
          key: "reasons3",
          value: "不想要了",
        },
        {
          key: "reasons4",
          value: "商品选错/多选",
        },
        {
          key: "reasons5",
          value: "地址信息填写错误",
        },
        {
          key: "reasons6",
          value: "商品降价",
        },
      ],
      otherReason: [
        {
          key: "other",
          value: "其它",
        },
      ],
    },
    "en-US": {
      basic: "Basic Usage",
      tipsText: "Components with warm tips",
      otherText: "Components with other reason options",
      cellTitle: "Show Dialog",
      checkboxText: "",
      cancelText: "",
      tipsTitle: "reminder",
      reasonTitle: "Please select the reason for canceling the order",
      cancelReasonTitle: "Refund reason",
      submitText: "confirm",
      textareaPlaceholder: "Please enter content",
      warmTips: [
        "1. Limited time special offers, reservation qualifications and other purchase privileges may be cancelled at the same time",
        "2. In case of order splitting, coupons will be exchanged for beans of the same value and returned",
        "3. The payment voucher will not be returned; Cancellation of payment preference",
        "4. Once the order is cancelled, it cannot be recovered",
      ],
      cancelReason: [
        {
          key: "reasons1",
          value: "No goods",
        },
        {
          key: "reasons2",
          value: "Delivery time problem",
        },
        {
          key: "reasons3",
          value: "do not want goods",
        },
        {
          key: "reasons4",
          value: "Wrong goods selected",
        },
        {
          key: "reasons5",
          value: "Incorrect address information",
        },
        {
          key: "reasons6",
          value: "Commodity price reduction",
        },
      ],
      otherReason: [
        {
          key: "other",
          value: "other",
        },
      ],
    },
  });

  //合并other其它原因
  const otherReasonList = React.useMemo(() => {
    return [...translated.cancelReason, ...translated.otherReason];
  }, []);
  //公共参数
  const buttonProps: Partial<ButtonProps> = React.useMemo(() => {
    return {
      type: "primary",
      className: "cancel-btn",
    };
  }, []);
  const textareaProps: Partial<TextAreaProps> = React.useMemo(() => {
    return {
      placeholder: translated.textareaPlaceholder,
      rows: "3",
      limitshow: true,
      maxlength: 100,
    };
  }, []);
  const popupTitleMemo = React.useMemo(() => {
    return translated.cancelReasonTitle;
  }, []);
  const reasonTitleMemo = React.useMemo(() => {
    return <div>{translated.reasonTitle}</div>;
  }, []);

  //基本使用
  const [showPanel, setShowPanel] = useState(false);
  //关闭弹窗触发的事件
  const clickClosePopUp = React.useCallback(() => {
    setShowPanel(false);
  }, [showPanel]);

  //带有温馨提示的组件
  const [showCancelPanel, setShowCancelPanel] = useState(false);
  const clickClosePopUpSec = React.useCallback(() => {
    setShowCancelPanel(false);
  }, [showCancelPanel]);

  //带有其它原因选项的组件
  const [showOtherCancelPanel, setShowOtherCancelPanel] = useState(false);
  const clickClosePopUpThree = React.useCallback(() => {
    setShowOtherCancelPanel(false);
  }, [showOtherCancelPanel]);

  //可以取消已选择原因选项的组件
  const [showCancelCancelPanel, setShowCancelCancelPanel] = useState(false);
  const clickClosePopUpCancel = React.useCallback(() => {
    setShowCancelCancelPanel(false);
  }, [showCancelCancelPanel]);

  // checkbox在前面
  const [showcheckboxCancelPanel, setShowcheckboxCancelPanel] = useState(false);
  const clickClosePopUpCheckbox = React.useCallback(() => {
    setShowcheckboxCancelPanel(false);
  }, [showcheckboxCancelPanel]);
  //提交事件
  const submitBtn = React.useCallback(
    (
      selectedReason: IreasonsObject,
      textAreaValue: string,
      switchStatus: boolean
    ) => {
      console.log(
        `selectedReason:${JSON.stringify(
          selectedReason
        )}, textAreaValue:${textAreaValue},switchStatus:${switchStatus}`
      );
      clickClosePopUp();
      clickClosePopUpSec();
      clickClosePopUpThree();
      clickClosePopUpCancel();
      clickClosePopUpCheckbox();
    },
    []
  );

  return (
    <>
      <div className="demo">
        <h2>{translated.basic}</h2>
        <Cell title={translated.cellTitle} onClick={() => setShowPanel(true)} />
        <OrderCancelPanel
          showCancelPanel={showPanel}
          popupTitle={popupTitleMemo}
          cancelReason={translated.cancelReason}
          buttonProps={buttonProps}
          onClose={clickClosePopUp}
          onSubmitBtn={submitBtn}
        />
        <h2>{translated.tipsText}</h2>
        <Cell
          title={translated.cellTitle}
          onClick={() => setShowCancelPanel(true)}
        />
        <OrderCancelPanel
          showCancelPanel={showCancelPanel}
          popupTitle={popupTitleMemo}
          reasonTitle={reasonTitleMemo}
          cancelReason={translated.cancelReason}
          warmTips={translated.warmTips}
          tipsTitle={translated.tipsTitle}
          submitText={translated.submitText}
          buttonProps={buttonProps}
          onClose={clickClosePopUpSec}
          onSubmitBtn={submitBtn}
        />
        <h2>{translated.otherText}</h2>
        <Cell
          title={translated.cellTitle}
          onClick={() => setShowOtherCancelPanel(true)}
        />
        <OrderCancelPanel
          showCancelPanel={showOtherCancelPanel}
          popupTitle={popupTitleMemo}
          tipsTitle={""}
          submitText={translated.submitText}
          warmTips={[]}
          cancelReason={otherReasonList}
          buttonProps={buttonProps}
          textAreaProps={textareaProps}
          onClose={clickClosePopUpThree}
          onSubmitBtn={submitBtn}
        />
        <h2>{translated.cancelText}</h2>
        <Cell
          title={translated.cellTitle}
          onClick={() => setShowCancelCancelPanel(true)}
        />
        <OrderCancelPanel
          showCancelPanel={showCancelCancelPanel}
          popupTitle={popupTitleMemo}
          canCancelReason={true}
          cancelReason={translated.cancelReason}
          buttonProps={buttonProps}
          textAreaProps={textareaProps}
          onClose={clickClosePopUpCancel}
          onSubmitBtn={submitBtn}
        />
        <h2>{translated.checkboxText}</h2>
        <Cell
          title={translated.cellTitle}
          onClick={() => setShowcheckboxCancelPanel(true)}
        />
        <OrderCancelPanel
          showCancelPanel={showcheckboxCancelPanel}
          checkboxType="front"
          showBtntips={true}
          popupTitle={popupTitleMemo}
          cancelReason={translated.cancelReason}
          buttonProps={buttonProps}
          textAreaProps={textareaProps}
          onClose={clickClosePopUpCheckbox}
          onSubmitBtn={submitBtn}
        />
      </div>
    </>
  );
};

export default CouponDemo;
