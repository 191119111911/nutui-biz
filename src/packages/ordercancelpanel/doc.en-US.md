# OrderCancelPanel

### Intro

OrderCancelPanel

### Install

```javascript
import { Cell } from "@nutui/nutui-react";
import { OrderCancelPanel } from "@nutui/nutui-biz";
```

## Demo

### Basic Usage

:::demo

```tsx
import React, { useState } from "react";
import { Cell, ButtonProps } from "@nutui/nutui-react";
import { OrderCancelPanel } from "@nutui/nutui-biz";

const App = () => {
  const cancelReason = [
    {
      key: "reasons1",
      value: "No goods",
    },
    {
      key: "reasons2",
      value: "Delivery time problem",
    },
    {
      key: "reasons3",
      value: "do not want goods",
    },
    {
      key: "reasons4",
      value: "Wrong goods selected",
    },
    {
      key: "reasons5",
      value: "Incorrect address information",
    },
    {
      key: "reasons6",
      value: "Commodity price reduction",
    },
  ];
  const clickClosePopUp = () => {
    setShowPanel(false);
  };
  const submitBtn = (currActivedKey: string, textAreaValue: string) => {
    console.log(
      `currActivedKey:${currActivedKey}, textAreaValue,${textAreaValue}`
    );
    setShowPanel(false);
  };
  const [showPanel, setShowPanel] = useState(false);
  const buttonProps: Partial<ButtonProps> = React.useMemo(() => {
    return {
      type: "primary",
      className: "cancel-btn",
    };
  }, []);
  return (
    <>
      <div className="demo">
        <Cell title="Basic Usage" onClick={() => setShowPanel(true)} />
        <OrderCancelPanel
          showCancelPanel={showPanel}
          popupTitle={<div>Refund reason</div>}
          cancelReason={cancelReason}
          buttonProps={buttonProps}
          onClickCloseIcon={clickClosePopUp}
          onClose={clickClosePopUp}
          onClickOverlay={clickClosePopUp}
          onSubmitBtn={submitBtn}
        />
      </div>
    </>
  );
};
export default App;
```

:::

### Components with warm tips

:::demo

```tsx
import React, { useState } from "react";
import { Cell, ButtonProps } from "@nutui/nutui-react";
import { OrderCancelPanel } from "@nutui/nutui-biz";

const App = () => {
  const warmTips = [
    "1. Limited time special offers, reservation qualifications and other purchase privileges may be cancelled at the same time",
    "2. In case of order splitting, coupons will be exchanged for beans of the same value and returned",
    "3. The payment voucher will not be returned; Cancellation of payment preference",
    "4. Once the order is cancelled, it cannot be recovered",
  ];
  const cancelReason = [
    {
      key: "reasons1",
      value: "No goods",
    },
    {
      key: "reasons2",
      value: "Delivery time problem",
    },
    {
      key: "reasons3",
      value: "do not want goods",
    },
    {
      key: "reasons4",
      value: "Wrong goods selected",
    },
    {
      key: "reasons5",
      value: "Incorrect address information",
    },
    {
      key: "reasons6",
      value: "Commodity price reduction",
    },
  ];
  const clickClosePopUp = () => {
    setShowCancelPanel(false);
  };
  const submitBtn = (currActivedKey: string, textAreaValue: string) => {
    console.log(
      `currActivedKey:${currActivedKey}, textAreaValue,${textAreaValue}`
    );
    setShowCancelPanel(false);
  };
  const [showCancelPanel, setShowCancelPanel] = useState(false);
  const buttonProps: Partial<ButtonProps> = React.useMemo(() => {
    return {
      type: "primary",
      className: "cancel-btn",
    };
  }, []);
  return (
    <>
      <div className="demo">
        <Cell
          title="Components with other reason options"
          onClick={() => setShowOtherCancelPanel(true)}
        />
        <OrderCancelPanel
          showCancelPanel={showOtherCancelPanel}
          popupTitle={popupTitleMemo}
          canCancelReason={true}
          tipsTitle="Refund reason"
          submitText="confirm"
          warmTips={warmTips}
          cancelReason={cancelReason}
          buttonProps={buttonProps}
          textAreaProps={textareaProps}
          onClickCloseIcon={clickClosePopUp}
          onClose={clickClosePopUp}
          onClickOverlay={clickClosePopUp}
          onSubmitBtn={submitBtn}
        />
      </div>
    </>
  );
};
export default App;
```

:::

### Components with other reason options

:::demo

```tsx
import React from "react";
import { Cell, ButtonProps, TextAreaProps } from "@nutui/nutui-react";
import { OrderCancelPanel } from "@nutui/nutui-biz";

const App = () => {
  const warmTips = [
    "1. Limited time special offers, reservation qualifications and other purchase privileges may be cancelled at the same time",
    "2. In case of order splitting, coupons will be exchanged for beans of the same value and returned",
    "3. The payment voucher will not be returned; Cancellation of payment preference",
    "4. Once the order is cancelled, it cannot be recovered",
  ];
  const cancelReason = [
    {
      key: "reasons1",
      value: "No goods",
    },
    {
      key: "reasons2",
      value: "Delivery time problem",
    },
    {
      key: "reasons3",
      value: "do not want goods",
    },
    {
      key: "reasons4",
      value: "Wrong goods selected",
    },
    {
      key: "reasons5",
      value: "Incorrect address information",
    },
    {
      key: "reasons6",
      value: "Commodity price reduction",
    },
    {
      key: "other",
      value: "other",
    },
  ];
  const clickClosePopUp = () => {
    setShowOtherCancelPanel(false);
  };
  const submitBtn = (currActivedKey: string, textAreaValue: string) => {
    console.log(
      `currActivedKey:${currActivedKey}, textAreaValue,${textAreaValue}`
    );
    setShowOtherCancelPanel(false);
  };
  const [showOtherCancelPanel, setShowOtherCancelPanel] = useState(false);
  const buttonProps: Partial<ButtonProps> = React.useMemo(() => {
    return {
      type: "primary",
      className: "cancel-btn",
    };
  }, []);
  const textareaProps: Partial<TextAreaProps> = React.useMemo(() => {
    return {
      placeholder: translated.textareaPlaceholder,
      rows: "3",
      limitshow: true,
      maxlength: 100,
    };
  }, []);
  const popupTitleMemo = React.useMemo(() => {
    return <div>Refund reason</div>;
  }, []);
  return (
    <>
      <div className="demo">
        <Cell
          title="Components with other reason options"
          onClick={() => setShowOtherCancelPanel(true)}
        />
        <OrderCancelPanel
          showCancelPanel={showOtherCancelPanel}
          popupTitle={popupTitleMemo}
          canCancelReason={true}
          tipsTitle="reminder"
          submitText="confirm"
          warmTips={warmTips}
          cancelReason={cancelReason}
          buttonProps={buttonProps}
          textAreaProps={textareaProps}
          onClickCloseIcon={clickClosePopUp}
          onClose={clickClosePopUp}
          onClickOverlay={clickClosePopUp}
          onSubmitBtn={submitBtn}
        />
      </div>
    </>
  );
};
export default App;
```

:::

## API

### Props

| Attribute       | Description                                          | Type                  | Default  |
| --------------- | ---------------------------------------------------- | --------------------- | -------- |
| showCancelPanel | Whether to display or cancel the order pop-up window | Boolean               | `false`  |
| warmTips        | Warm prompt content, no prompt content               | Array<string>         | --       |
| cancelReason    | Reason for cancellation                              | Array<IreasonsObject> | --       |
| canCancelReason | Click again to cancel the selected reason            | Boolean               | `false`  |
| popupTitle      | Main title of pop-up window                          | ReactNode             | --       |
| reasonTitle     | Title of cancellation reason                         | ReactNode             | --       |
| submitText      | Popup Button content                                 | string                | `submit` |
| tipsTitle       | Warm prompt title                                    | string                | --       |
| buttonProps     | Props of button components                           | ButtonProps           | --       |
| textAreaProps   | props of TextAre                                     | TextAreaProps         | --       |

### IreasonsObject

| Attribute | Description                         | Type   | Default |
| --------- | ----------------------------------- | ------ | ------- |
| key       | Key of cancellation reason          | string | --      |
| value     | Copy content of cancellation reason | string | --      |

### Events

| Attribute        | Description                                      | Arguments |
| ---------------- | ------------------------------------------------ | --------- |
| onClose          | Triggered when clicking the pop-up box           | --        |
| onClickCloseIcon | Triggered when the close icon is clicked         | --        |
| onClickOverlay   | Click the mask to trigger                        | --        |
| onSubmitBtn      | Click the submit button to trigger The parameter | --        |
