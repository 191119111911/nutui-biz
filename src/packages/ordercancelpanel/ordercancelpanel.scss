@import "../../styles/mixins/text-ellipsis.scss";
.nut-biz-ordercancel {
  font-weight: normal;
  &__main {
    padding: 0px 20px;
  }
  &__reason-header {
    margin-top: 10px;
    width: 90%;
    font-size: 18px;
    @include text-ellipsis;
  }
  &__header {
    width: 90%;
    text-align: left;
    font-size: 18px;
    height: 50px;
    line-height: 50px;
    @include text-ellipsis;
  }
  &__tips {
    font-size: 14px;
    color: #000000a6;
    padding: 14px;
    background-color: #f7f7f7ff;
    line-height: 25px;
    border-radius: 10px;
    max-height: 160px;
    overflow: auto;
    margin-bottom: 10px;
    &-header {
      font-size: 14px;
      text-align: left;
      font-weight: normal;
    }
  }
  &__reason {
    font-size: 14px;
    max-height: 280px;
    overflow: auto;
    &--list {
      display: flex;
      height: 35px;
      align-items: center;
      justify-content: space-between;
      position: relative;
      &__label {
        // height: 35px;
        // line-height: 35px;
        width: 100%;
        @include text-ellipsis;
      }
      &__checkbox {
        &--post-front {
          margin-right: 10px;
        }
        .nut-checkbox {
          display: flex;
          justify-content: right;
          .nut-checkbox__label {
            display: none;
          }
        }
      }
      &__overlay {
        position: absolute;
        top: 0px;
        left: 0px;
        height: 35px;
        width: 100%;
        opacity: 0;
      }
    }
  }
  &__area {
    border: 1px solid #eee;
  }
  &__btns {
    margin-top: 10px;
    width: 100%;
    box-shadow: 1px 14px 43px -7px rgb(0 0 0 / 50%);
    &--tips {
      font-size: 14px;
      display: flex;
      padding: 20px 20px 0px 20px;
      justify-content: space-between;
    }
    &--button {
      height: 75px;
      display: flex;
      justify-content: center;
      align-items: center;
    }
    &--tap {
      height: 30px;
    }
  }
  .cancel-btn {
    width: 80%;
  }
}
