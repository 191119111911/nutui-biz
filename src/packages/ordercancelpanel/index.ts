import { OrderCancelPanel } from './ordercancelpanel'

export default OrderCancelPanel
