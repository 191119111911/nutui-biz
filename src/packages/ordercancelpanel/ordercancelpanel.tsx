import React, {
  FunctionComponent,
} from 'react'
import { useConfig } from '@/packages/configprovider'

import { IComponent } from '@/utils/typings'

export interface OrderCancelPanelProps extends IComponent {
}

const defaultProps = {
} as OrderCancelPanelProps

export const OrderCancelPanel: FunctionComponent<
  Partial<OrderCancelPanelProps> & Omit<React.HTMLAttributes<HTMLDivElement>, 'onChange'>
> = (props) => {
  const { locale } = useConfig()
  const {
  } = {
    ...defaultProps,
    ...props,
  }

  return (
    <div>
    </div>
  )
}

OrderCancelPanel.defaultProps = defaultProps
OrderCancelPanel.displayName = 'NutOrderCancelPanel'
