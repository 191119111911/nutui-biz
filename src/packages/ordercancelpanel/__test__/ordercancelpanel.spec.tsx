import * as React from "react";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import { OrderCancelPanel } from "../ordercancelpanel";

test("should match snapshot", () => {
  const { asFragment } = render(
    <>
      <OrderCancelPanel
        showCancelPanel={true}
        popupTitle="退款原因"
        cancelReason={[
          {
            key: "reasons1",
            value: "商品无货",
          },
          {
            key: "reasons2",
            value: "发货时间问题",
          },
          {
            key: "reasons3",
            value: "不想要了",
          },
          {
            key: "reasons4",
            value: "商品选错/多选",
          },
          {
            key: "reasons5",
            value: "地址信息填写错误",
          },
          {
            key: "reasons6",
            value: "商品降价",
          },
        ]}
        buttonProps={{
          type: "primary",
          className: "cancel-btn",
        }}
      />
    </>
  );
  expect(asFragment()).toMatchSnapshot();
});

test("prop of Basic Usage", () => {
  const closeClick = jest.fn();
  const state = {
    popupTitle: "退款原因",
    cancelReason: [
      {
        key: "reasons1",
        value: "商品无货",
      },
      {
        key: "reasons2",
        value: "发货时间问题",
      },
      {
        key: "reasons3",
        value: "不想要了",
      },
      {
        key: "reasons4",
        value: "商品选错/多选",
      },
      {
        key: "reasons5",
        value: "地址信息填写错误",
      },
      {
        key: "reasons6",
        value: "商品降价",
      },
    ],
  };
  const { container } = render(
    <OrderCancelPanel
      showCancelPanel={true}
      popupTitle={state.popupTitle}
      cancelReason={state.cancelReason}
      onSubmitBtn={closeClick}
      buttonProps={{
        type: "primary",
        className: "cancel-btn",
      }}
    />
  );
  expect(
    container.querySelector(".nb-ordercancel .nb-ordercancel__header")
  ).toHaveTextContent("退款原因");

  expect(
    container.querySelectorAll(
      ".nb-ordercancel .nb-ordercancel__reason .nb-ordercancel__reason--list"
    ).length
  ).toBe(state.cancelReason.length);

  expect(
    container.querySelector(".nb-ordercancel .nb-ordercancel__btns .nut-button")
  ).toHaveClass("nut-button--primary");

  //点击了某个原因
  const nutOrderCancelList = container
    .querySelectorAll(
      ".nb-ordercancel .nb-ordercancel__main .nb-ordercancel__reason .nb-ordercancel__reason--list"
    )[1]
    .querySelector(".nb-ordercancel__reason--list__overlay") as HTMLElement;
  fireEvent.click(nutOrderCancelList);
  expect(
    container
      .querySelectorAll(
        ".nb-ordercancel .nb-ordercancel__main .nb-ordercancel__reason .nb-ordercancel__reason--list"
      )[1]
      .querySelector(
        ".nb-ordercancel__reason--list__checkbox .nut-checkbox .nut-icon"
      )
  ).toHaveClass("nut-icon-checked");

  //点击了关闭icon按钮
  const nutOrderCancelCloseIcon = container.querySelector(
    ".nb-ordercancel .nb-ordercancel__btns .nb-ordercancel__btns--button .nut-button"
  ) as HTMLElement;
  fireEvent.click(nutOrderCancelCloseIcon);
  expect(closeClick).toBeCalled();
});

test("Components with warm tips", () => {
  const state = {
    popupTitle: "退款原因",
    warmTips: [
      "1. 限时特价、预约资格等购买优惠可能一并取消",
      "2. 如遇订单拆分，京券将换成同价值京豆返还",
      "3. 支付券不予返还；支付优惠一并取消",
      "4. 订单一旦取消，无法恢复",
    ],
    cancelReason: [
      {
        key: "reasons1",
        value: "商品无货",
      },
      {
        key: "reasons2",
        value: "发货时间问题",
      },
      {
        key: "reasons3",
        value: "不想要了",
      },
      {
        key: "reasons4",
        value: "商品选错/多选",
      },
      {
        key: "reasons5",
        value: "地址信息填写错误",
      },
      {
        key: "reasons6",
        value: "商品降价",
      },
    ],
  };
  const { container } = render(
    <OrderCancelPanel
      warmTips={state.warmTips}
      showCancelPanel={true}
      popupTitle={state.popupTitle}
      cancelReason={state.cancelReason}
      canCancelReason={true}
      buttonProps={{
        type: "primary",
        className: "cancel-btn",
      }}
    />
  );
  expect(
    container.querySelector(".nb-ordercancel .nb-ordercancel__header")
  ).toHaveTextContent("退款原因");
  expect(
    container.querySelectorAll(
      ".nb-ordercancel .nb-ordercancel__reason .nb-ordercancel__reason--list"
    ).length
  ).toBe(state.cancelReason.length);
  expect(
    container.querySelectorAll(
      ".nb-ordercancel .nb-ordercancel__tips .nb-ordercancel__tips-list"
    ).length
  ).toBe(state.warmTips.length);
  expect(
    container.querySelector(".nb-ordercancel .nb-ordercancel__btns .nut-button")
  ).toHaveClass("nut-button--primary");

  const nutOrderCancelList = container
    .querySelectorAll(
      ".nb-ordercancel .nb-ordercancel__reason .nb-ordercancel__reason--list"
    )[1]
    .querySelector(".nb-ordercancel__reason--list__overlay") as HTMLElement;
  fireEvent.click(nutOrderCancelList);
  expect(
    container
      .querySelectorAll(
        ".nb-ordercancel .nb-ordercancel__reason .nb-ordercancel__reason--list"
      )[1]
      .querySelector(
        ".nb-ordercancel__reason--list__checkbox .nut-checkbox .nut-icon"
      )
  ).toHaveClass("nut-icon-checked");
  fireEvent.click(nutOrderCancelList);
  expect(
    container
      .querySelectorAll(
        ".nb-ordercancel .nb-ordercancel__reason .nb-ordercancel__reason--list"
      )[1]
      .querySelector(
        ".nb-ordercancel__reason--list__checkbox .nut-checkbox .nut-icon"
      )
  ).toHaveClass("nut-icon-check-normal");
});

test("Components with other reason options", () => {
  const onClick1 = jest.fn();
  const state = {
    popupTitle: "退款原因",
    warmTips: [
      "1. 限时特价、预约资格等购买优惠可能一并取消",
      "2. 如遇订单拆分，京券将换成同价值京豆返还",
      "3. 支付券不予返还；支付优惠一并取消",
      "4. 订单一旦取消，无法恢复",
    ],
    cancelReason: [
      {
        key: "reasons1",
        value: "商品无货",
      },
      {
        key: "reasons2",
        value: "发货时间问题",
      },
      {
        key: "reasons3",
        value: "不想要了",
      },
      {
        key: "reasons4",
        value: "商品选错/多选",
      },
      {
        key: "reasons5",
        value: "地址信息填写错误",
      },
      {
        key: "reasons6",
        value: "商品降价",
      },
      {
        key: "other",
        value: "其他",
      },
    ],
  };
  const { container } = render(
    <OrderCancelPanel
      warmTips={state.warmTips}
      showCancelPanel={true}
      popupTitle={state.popupTitle}
      cancelReason={state.cancelReason}
      onSubmitBtn={onClick1}
      buttonProps={{
        type: "primary",
        className: "cancel-btn",
      }}
    />
  );
  //点击其他选项
  const nutOrderCancelList = container
    .querySelectorAll(
      ".nb-ordercancel .nb-ordercancel__reason .nb-ordercancel__reason--list"
    )
    [state.cancelReason.length - 1].querySelector(
      ".nb-ordercancel__reason--list__overlay"
    ) as HTMLElement;
  fireEvent.click(nutOrderCancelList);
  const textareaEl = container.querySelector(
    ".nb-ordercancel .nb-ordercancel__reason .nut-textarea"
  ) as HTMLTextAreaElement;
  expect(textareaEl).toHaveClass("nb-ordercancel__area");

  //文字改变
  const textareaDom = container.querySelector(
    ".nb-ordercancel .nb-ordercancel__reason .nut-textarea .nut-textarea__textarea"
  ) as HTMLTextAreaElement;
  fireEvent.change(textareaDom, { target: { value: "文字改变" } });
  expect(textareaDom.innerHTML).toBe("文字改变");
  //点击第一个
  const nutOrderCancelListFirst = container
    .querySelectorAll(
      ".nb-ordercancel .nb-ordercancel__reason .nb-ordercancel__reason--list"
    )[0]
    .querySelector(".nb-ordercancel__reason--list__overlay") as HTMLElement;
  fireEvent.click(nutOrderCancelListFirst);

  //点击确认按钮
  const nutOrderCancelBtns = container.querySelector(
    ".nb-ordercancel .nb-ordercancel__btns .nut-button--normal"
  ) as HTMLElement;
  fireEvent.click(nutOrderCancelBtns);
  expect(onClick1).toBeCalled();
});
