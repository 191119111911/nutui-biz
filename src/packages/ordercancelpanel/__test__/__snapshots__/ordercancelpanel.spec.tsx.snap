// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="overlay-fade-enter-active nut-overlay"
    style="z-index: 1000; animation-duration: 0.3s;"
  />
  <div
    class="round nut-popup-bottom  nb-ordercancel undefined  nut-popup"
    style="z-index: 1000; animation-duration: 0.3s;"
  >
    <div
      class="nb-ordercancel__main"
    >
      <div
        class="nb-ordercancel__header"
      >
        退款原因
      </div>
      <div
        class="nb-ordercancel__reason"
      >
        <div
          class="nb-ordercancel__reason--list"
        >
          <div
            class="nb-ordercancel__reason--list__label"
          >
            商品无货
          </div>
          <div
            class="nb-ordercancel__reason--list__checkbox"
          >
            <div
              class="nut-checkbox "
            >
              <i
                class="nutui-iconfont nut-icon nut-icon-check-normal nut-checkbox__icon--unchecked"
                style="font-size: 18px; height: 18px; width: 18px;"
              />
              <span
                class="nut-checkbox__label "
              />
            </div>
          </div>
          <div
            class="nb-ordercancel__reason--list__overlay"
          />
        </div>
        <div
          class="nb-ordercancel__reason--list"
        >
          <div
            class="nb-ordercancel__reason--list__label"
          >
            发货时间问题
          </div>
          <div
            class="nb-ordercancel__reason--list__checkbox"
          >
            <div
              class="nut-checkbox "
            >
              <i
                class="nutui-iconfont nut-icon nut-icon-check-normal nut-checkbox__icon--unchecked"
                style="font-size: 18px; height: 18px; width: 18px;"
              />
              <span
                class="nut-checkbox__label "
              />
            </div>
          </div>
          <div
            class="nb-ordercancel__reason--list__overlay"
          />
        </div>
        <div
          class="nb-ordercancel__reason--list"
        >
          <div
            class="nb-ordercancel__reason--list__label"
          >
            不想要了
          </div>
          <div
            class="nb-ordercancel__reason--list__checkbox"
          >
            <div
              class="nut-checkbox "
            >
              <i
                class="nutui-iconfont nut-icon nut-icon-check-normal nut-checkbox__icon--unchecked"
                style="font-size: 18px; height: 18px; width: 18px;"
              />
              <span
                class="nut-checkbox__label "
              />
            </div>
          </div>
          <div
            class="nb-ordercancel__reason--list__overlay"
          />
        </div>
        <div
          class="nb-ordercancel__reason--list"
        >
          <div
            class="nb-ordercancel__reason--list__label"
          >
            商品选错/多选
          </div>
          <div
            class="nb-ordercancel__reason--list__checkbox"
          >
            <div
              class="nut-checkbox "
            >
              <i
                class="nutui-iconfont nut-icon nut-icon-check-normal nut-checkbox__icon--unchecked"
                style="font-size: 18px; height: 18px; width: 18px;"
              />
              <span
                class="nut-checkbox__label "
              />
            </div>
          </div>
          <div
            class="nb-ordercancel__reason--list__overlay"
          />
        </div>
        <div
          class="nb-ordercancel__reason--list"
        >
          <div
            class="nb-ordercancel__reason--list__label"
          >
            地址信息填写错误
          </div>
          <div
            class="nb-ordercancel__reason--list__checkbox"
          >
            <div
              class="nut-checkbox "
            >
              <i
                class="nutui-iconfont nut-icon nut-icon-check-normal nut-checkbox__icon--unchecked"
                style="font-size: 18px; height: 18px; width: 18px;"
              />
              <span
                class="nut-checkbox__label "
              />
            </div>
          </div>
          <div
            class="nb-ordercancel__reason--list__overlay"
          />
        </div>
        <div
          class="nb-ordercancel__reason--list"
        >
          <div
            class="nb-ordercancel__reason--list__label"
          >
            商品降价
          </div>
          <div
            class="nb-ordercancel__reason--list__checkbox"
          >
            <div
              class="nut-checkbox "
            >
              <i
                class="nutui-iconfont nut-icon nut-icon-check-normal nut-checkbox__icon--unchecked"
                style="font-size: 18px; height: 18px; width: 18px;"
              />
              <span
                class="nut-checkbox__label "
              />
            </div>
          </div>
          <div
            class="nb-ordercancel__reason--list__overlay"
          />
        </div>
      </div>
    </div>
    <div
      class="nb-ordercancel__btns"
    >
      <div
        class="nb-ordercancel__btns--button"
      >
        <button
          class="nut-button nut-button--primary nut-button--normal nut-button--round cancel-btn"
        >
          <div
            class="nut-button__warp"
          >
            <div
              class=""
            >
              提交
            </div>
          </div>
        </button>
      </div>
    </div>
    <div
      class="nut-popup__close-icon nut-popup__close-icon--top-right"
    >
      <i
        class="nutui-iconfont nut-icon nut-icon-close "
        style="font-size: 12px; height: 12px; width: 12px;"
      />
    </div>
  </div>
</DocumentFragment>
`;
